cookbooks/aws-parallelcluster-environment/recipes/init/backup_home_shared_data.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return if on_docker?
+
+if node['cluster']['node_type'] == 'HeadNode'
+  # For each, backup the data to a temp location
+  # This is necessary to preserve any data in these directories that was
+  # generated during the build of ParallelCluster AMIs after converting to
+  # shared storage
+  bash "Backup /home" do
+    user 'root'
+    group 'root'
+    code <<-EOH
+      mkdir -p /tmp/home
```

Comment:
Security best practice: use random names for temporary folders to prevent attacks based on name clashing. Eg: better to use `mktemp -d` rather than `mkdir`

Reply:
How would an attacker be able to leverage this directory?  I'm not familiar with a name clashing attack

Reply:
It's a scenario where an attacker compromises your data leveraging the fact that you temporarily put it in a less restrictive folder with a well known name and then copy back from there. If you use random name, it's harder for the attacker to know where to look at. The mitigations are:
1. restrict permissions to the temporary folder (the most important mitigation)
2. random names for temporary folders
3. minimize the life time of temporary data, i.e. explicitly remove it after being used)

Reply:
This is not a blocking comment because the temp folder is created as root, so by default the folder has write permissions granted only to root (mitigation 1)

Reply:
got it thanks!